# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/abiword/hotfix-eds.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Use e_vcard_convert_to_string() with evolution-data-server >= 3.60.

Copyright (C) 2026 René Rebe <rene@exactco.de>
Signed-off-by: René Rebe <rene@exactco.de>

--- AbiWord-release-3.0.8.orig/src/text/ptbl/xp/pd_DocumentRDF.cpp	2026-09-27 19:18:45.821995497 +0200
+++ AbiWord-release-3.0.8/src/text/ptbl/xp/pd_DocumentRDF.cpp	2026-09-27 19:18:45.837091118 +0200
@@ -2034,7 +2034,11 @@ PD_RDFContact::exportToFile( const std::
         set( c, EVC_TEL,      m_phone );
         set( c, EVC_X_JABBER, m_jabberID );
 
+#if EDS_CHECK_VERSION(3, 60, 0)
+        gchar* data =  e_vcard_convert_to_string( c, E_VCARD_VERSION_30 );
+#else
         gchar* data =  e_vcard_to_string( c, EVC_FORMAT_VCARD_30 );
+#endif
         UT_DEBUGMSG(( "saving vcard to file:%s vcard.len:%ld\n", filename.c_str(), strlen(data) ));
         std::ofstream oss( filename.c_str() );
         oss.write( data, strlen(data) );
